src/shims/unix/fs.rs

```diff
@@ -118,7 +118,7 @@ impl UnixFileDescription for FileHandle {
 
 impl<'tcx> EvalContextExtPrivate<'tcx> for crate::MiriInterpCx<'tcx> {}
 trait EvalContextExtPrivate<'tcx>: crate::MiriInterpCxExt<'tcx> {
-    fn macos_fbsd_solarish_write_stat_buf(
```

Comment:
I said in the issue that you have to check whether Linux has any fields in its `stat` that we need to add here, on top of the "basics". Did you do that? It seems like you did not, because there *are* extra fields: `st_mtime_nsec` and friends.